File: gcdset.py
import math
from functools import reduce

def gcd_list(numbers):
    return reduce(math.gcd, numbers)

def process_queries(queries):
    results = []
    
    for query in queries:
        n, k, a = query
        overall_gcd = gcd_list(a)
        
        if k % overall_gcd == 0:
            results.append("YES")
        else:
            results.append("NO")
    
    return results

# Input processing
T = int(input().strip())
queries = []

for _ in range(T):
    n, k = map(int, input().strip().split())
    a = list(map(int, input().strip().split()))
    queries.append((n, k, a))

# Process and print the results
results = process_queries(queries)
for result in results:
    print(result)

File: gcdset.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long ll;

int query() {
    ll n, k;
    cin >> n >> k;

    ll res = 0;
    for (int i = 0; i < n; ++i) {
        ll x;
        cin >> x;
        if (x % k == 0)
            res = __gcd(res, x / k);
    }

    return cout << (res == 1 ? "YES\n" : "NO\n"), 0;
}
int main () {
    int x;
    cin >> x;
    while(x--) {
        query();
    }
}